File: selection_sort.c
#include<stdio.h>
int i;
int j;
void selectionsort(int n , int arr[]){
    for(i=0;i<n-1;i++){
        int min = i;
     for(j=i+1;j<n;j++){
        if(arr[min] > arr[j]){
            min = j;
        }
     }

     if(min !=i){
        int temp = arr[min];
        arr[min] = arr[i];
        arr[i] = temp;

     }

    }
}
void printarr(int arr[],int n){
    for(i=0;i<n;i++){
        printf("%d",arr[i]);
    }
    printf("\n");
}
void main()
{
    int n;
    printf("Enter the number of elements in an array : ");
    scanf("%d", &n);
    int arr[n];

    printf("Enter the %dth element in an array : ", i+1);
    scanf("%d", arr[i]);
    printf("The Unsorted Array is : ");
    printf("\n");
    printarr( arr, n);

    selectionsort(n,arr);
    printf("The Sorted Array is : ");
    printf("\n");
     printarr( arr, n);
}

File: Selection_sort.java
import java.util.*;

public class Selection_sort{
    public static void selection_sort(int n,int []arr){
        for(int i=0;i<n-1;i++){
            int min =i;
        for(int j=i+1;j<n;j++){
            if(arr[min]>arr[j]){
                min =j;
            }
        }
        if(min !=i){
            int temp = arr[min];
            arr[min] = arr[i];
            arr[i]= temp;
        }
        }

    }

    public static void printArr(int[] arr)
    {
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the number of terms in an array = ");
        
        int n = sc.nextInt();
        int i;
        int [] arr = new int[n];
        for(i=0;i<n;i++){
            System.out.print("Enter the " +(i+1)+ " th in an array = ");
            arr[i] = sc.nextInt();
        
    }
    
   System.out.println("The Unsorted array is :");
   printArr(arr);
   selection_sort(n, arr);
   System.out.println();
   System.out.println("The Sorted Array is :");
   printArr(arr);
    }
    
}
